src/use_i18n.rs

```diff
@@ -146,26 +198,213 @@ impl I18n {
     }
 
     /// Get the selected language.
-    pub fn language(&mut self) -> LanguageIdentifier {
+    pub fn language(&self) -> LanguageIdentifier {
         self.selected_language.read().clone()
     }
 
     /// Get the fallback language.
-    pub fn fallback_language(&mut self) -> Option<LanguageIdentifier> {
+    pub fn fallback_language(&self) -> Option<LanguageIdentifier> {
         self.fallback_language.read().clone()
     }
 
     /// Update the selected language.
     pub fn set_language(&mut self, id: LanguageIdentifier) {
         *self.selected_language.write() = id;
+        self.update_active_bundle();
     }
 
     /// Update the fallback language.
     pub fn set_fallback_language(&mut self, id: LanguageIdentifier) {
         *self.fallback_language.write() = Some(id);
+        self.update_active_bundle();
+    }
+
+    fn update_active_bundle(&mut self) {
+        let bundle = create_bundle(
+            &self.selected_language.read(),
+            &self.fallback_language.read(),
+            &self.locale_resources.read(),
+            &self.locales.read(),
```

Comment:
`.read()` subscribes by default the current scope to any change in any of these signals, `.peek()` doesn't 

```suggestion
            &self.selected_language.peek(),
            &self.fallback_language.peek(),
            &self.locale_resources.peek(),
            &self.locales.peek(),
```